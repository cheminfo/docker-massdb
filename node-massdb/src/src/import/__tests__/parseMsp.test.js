'use strict';

let join = require('path').join;

let parseMsp = require('../parseMsp');

test('Test Msp parsing', async () => {
  let file = join(__dirname, 'data/nist.msp');
  let result = await parseMsp(file);
  expect(Object.keys(result)).toHaveLength(2);
  expect(result).toMatchSnapshot();
});
