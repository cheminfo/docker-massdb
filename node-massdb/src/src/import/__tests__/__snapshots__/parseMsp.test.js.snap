// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Msp parsing 1`] = `
Array [
  Object {
    "comments": "American Petroleum Institute Research Project 44",
    "data": Object {
      "x": Array [
        1,
        2,
      ],
      "y": Array [
        21,
        999,
      ],
    },
    "formula": "H2",
    "name": "Hydrogen",
    "nist": "245692",
    "rn": "1333-74-0",
    "synonyms": Array [
      "Dihydrogen",
      "o-Hydrogen",
      "p-Hydrogen",
      "Molecular hydrogen",
      "H2",
      "UN 1049",
      "UN 1966",
    ],
  },
  Object {
    "comments": "D.HENNEBERG, MAX-PLANCK INSTITUTE, MULHEIM, WEST GERMANY",
    "data": Object {
      "x": Array [
        2,
        4,
      ],
      "y": Array [
        15,
        999,
      ],
    },
    "formula": "D2",
    "name": "Deuterium",
    "nist": "61316",
    "rn": "7782-39-0",
    "synonyms": Array [
      "D2",
      "UN 1957",
    ],
  },
]
`;
